pkg/scheduler/plugins/coescheduling/core/core.go

```diff
@@ -0,0 +1,618 @@
+	for i := range pods {
+		gangIdTmp := GetNamespaceSplicingName(pods[i].Namespace,
+			GetGangNameByPod(pods[i]))
+		if gangIdTmp == gangId {
+			if pods[i].UID == pod.UID {
+				continue
+			}
+			toActivePods = append(toActivePods, pods[i])
+		}
+	}
+
+	if len(toActivePods) != 0 {
+		if c, err := state.Read(framework.PodsToActivateKey); err == nil {
+			if s, ok := c.(*framework.PodsToActivate); ok {
+				s.Lock()
+				for _, pod := range toActivePods {
+					namespacedName := GetNamespaceSplicingName(pod.Namespace, pod.Name)
+					s.Map[namespacedName] = pod
+				}
+				s.Unlock()
+			}
+		}
+	}
+}
+
+// PreFilter
+// if non-strict-mode, we only do step1 and step2:
+// i.Check whether childes in Gang has met the requirements of minimum number under each Gang, and reject the pod if negative.
+// ii.Check whether the Gang has been timeout(check the pod's annotation,later introduced at Permit section) or is inited, and reject the pod if positive.
+// iii.Check whether the Gang has met the scheduleCycleValid check, and reject the pod if negative.
+// iv.Try update scheduleCycle, scheduleCycleValid, childrenScheduleRoundMap as mentioned above.
+func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, pod *corev1.Pod) error {
+	gangName := GetGangNameByPod(pod)
+	if gangName == "" {
+		return nil
+	}
+	gangId := GetNamespaceSplicingName(pod.Namespace,
+		gangName)
+	gang := pgMgr.cache.getGangFromCache(gangId)
+	if gang == nil {
+		return fmt.Errorf("can't find gang, gangName: %v, podName: %v", gangId,
+			GetNamespaceSplicingName(pod.Namespace, pod.Name))
+	}
+	// first try update the global cycle of gang
+	gang.trySetScheduleCycleTrue()
+
+	gangScheduleCycle := gang.getGangScheduleCycle()
+	defer func() {
+		gang.setChildScheduleCycle(pod, gangScheduleCycle)
+	}()
+	// check if gang is inited
+	if !gang.HasGangInit {
+		return fmt.Errorf("gang has not init, gangName: %v, podName: %v", gangId,
+			GetNamespaceSplicingName(pod.Namespace, pod.Name))
+	}
+	// check if gang is timeout by pod's Annotation
+	// todo: if not annotate, how can we handle the timeout event
+	if pod.Annotations[extension.AnnotationGangTimeout] == "true" {
+		return fmt.Errorf("gang has timeout, gangName: %v, podName: %v", gangId,
+			GetNamespaceSplicingName(pod.Namespace, pod.Name))
+	}
+
+	if gang.getChildrenNum() < gang.getGangMinNum() {
+		return fmt.Errorf("gang child pod not collect enough, gangName: %v, podName: %v", gangId,
+			GetNamespaceSplicingName(pod.Namespace, pod.Name))
+	}
+	// if gang is created by PodGroup, we check the cluster resource as Scheduler-Plugins does
+	if !PgFromAnnotation(pod) {
+		// TODO(cwdsuzhou): This resource check may not always pre-catch unschedulable pod group.
+		// It only tries to PreFilter resource constraints so even if a PodGroup passed here,
+		// it may not necessarily pass Filter due to other constraints such as affinity/taints.
+		nodes, err := pgMgr.snapshotSharedLister.NodeInfos().List()
```

Comment:
this check is too simple that I think you can delete it. because there are so many reason to make schedule fail.

Reply:
ok, delete the resource check logic